data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -271,7 +271,7 @@ private SerializedJson getDocument(final Event event) {
     String docId = (documentIdField != null) ? event.get(documentIdField, String.class) : null;
     String routing = (routingField != null) ? event.get(routingField, String.class) : null;
 
-    final String document = DocumentBuilder.build(event, documentRootKey);
+    final String document = DocumentBuilder.build(event, documentRootKey, sinkContext().getTagsKeyName());
```

Comment:
I think we should add `SinkContext` to the constructor:

```
@DataPrepperPluginConstructor
public OpenSearchSink(final PluginSetting pluginSetting,
    final PluginFactory pluginFactory,
    final SinkContext sinkContext)
```

This will work with other sinks that have already migrated off of the `PluginSettings` model. In particular, `file` and `s3`.